math/mathcore/inc/Math/CladDerivator.h

```diff
@@ -241,6 +241,166 @@ ValueAndPushforward<Double_t, Double_t> Ln10_pushforward()
+      _d_v += (p3[2] + p3[3] * v) * d_out / _t12 * v;
+      _d_v += (p3[1] + (p3[2] + p3[3] * v) * v) * d_out / _t12;
+      double _r11 = d_out * -((p3[0] + (p3[1] + (p3[2] + p3[3] * v) * v) * v) / (_t12 * _t12));
+      _d_v += q3[3] * _r11 * v * v;
+      _d_v += (q3[2] + q3[3] * v) * _r11 * v;
+      _d_v += (q3[1] + (q3[2] + q3[3] * v) * v) * _r11;
+   } else if (v < 12) {
+      double _d_u = 0;
+      double u = 1. / v;
+      double _t15 = (q4[0] + (q4[1] + (q4[2] + q4[3] * u) * u) * u);
+      _d_u += p4[3] * d_out / _t15 * u * u;
+      _d_u += (p4[2] + p4[3] * u) * d_out / _t15 * u;
+      _d_u += (p4[1] + (p4[2] + p4[3] * u) * u) * d_out / _t15;
+      double _r13 = d_out * -((p4[0] + (p4[1] + (p4[2] + p4[3] * u) * u) * u) / (_t15 * _t15));
+      _d_u += q4[3] * _r13 * u * u;
+      _d_u += (q4[2] + q4[3] * u) * _r13 * u;
+      _d_u += (q4[1] + (q4[2] + q4[3] * u) * u) * _r13;
+      double _r12 = _d_u * -(1. / (v * v));
+      _d_v += _r12;
+   } else if (v < 50) {
+      double _d_u = 0;
+      double u = 1. / v;
+      double _t18 = (q5[0] + (q5[1] + (q5[2] + q5[3] * u) * u) * u);
+      _d_u += p5[3] * d_out / _t18 * u * u;
+      _d_u += (p5[2] + p5[3] * u) * d_out / _t18 * u;
+      _d_u += (p5[1] + (p5[2] + p5[3] * u) * u) * d_out / _t18;
+      double _r15 = d_out * -((p5[0] + (p5[1] + (p5[2] + p5[3] * u) * u) * u) / (_t18 * _t18));
+      _d_u += q5[3] * _r15 * u * u;
+      _d_u += (q5[2] + q5[3] * u) * _r15 * u;
+      _d_u += (q5[1] + (q5[2] + q5[3] * u) * u) * _r15;
+      double _r14 = _d_u * -(1. / (v * v));
+      _d_v += _r14;
+   } else if (v < 300) {
+      double _d_u = 0;
+      double u = 1. / v;
+      double _t21 = (q6[0] + (q6[1] + (q6[2] + q6[3] * u) * u) * u);
+      _d_u += p6[3] * d_out / _t21 * u * u;
+      _d_u += (p6[2] + p6[3] * u) * d_out / _t21 * u;
+      _d_u += (p6[1] + (p6[2] + p6[3] * u) * u) * d_out / _t21;
+      double _r17 = d_out * -((p6[0] + (p6[1] + (p6[2] + p6[3] * u) * u) * u) / (_t21 * _t21));
+      _d_u += q6[3] * _r17 * u * u;
+      _d_u += (q6[2] + q6[3] * u) * _r17 * u;
+      _d_u += (q6[1] + (q6[2] + q6[3] * u) * u) * _r17;
+      double _r16 = _d_u * -(1. / (v * v));
+      _d_v += _r16;
+   } else {
+      double _d_u = 0;
+      double _t25 = ::std::log(v);
+      double _t24 = (v + 1);
+      double _t23 = (v - v * _t25 / _t24);
+      double u = 1. / _t23;
+      double _t26;
+      _d_u += a2[3] * -d_out * u * u;
+      _d_u += (a2[2] + a2[3] * u) * -d_out * u;
+      _d_u += (a2[1] + (a2[2] + a2[3] * u) * u) * -d_out;
+      double _r18 = _d_u * -(1. / (_t23 * _t23));
+      _d_v += _r18;
+      _d_v += -_r18 / _t24 * _t25;
+      double _r19 = 0;
+      _r19 += v * -_r18 / _t24 * clad::custom_derivatives::log_pushforward(v, 1.).pushforward;
+      _d_v += _r19;
+      double _r20 = -_r18 * -(v * _t25 / (_t24 * _t24));
+      _d_v += _r20;
+   }
+
+   *d_x += _d_v / xi;
+   *d_x0 += -_d_v / xi;
+   *d_xi += _d_v * -((x - x0) / (xi * xi));
+}
```

Comment:
@petrozarytskyi, @vaithak, you take a look and maybe simplify this code?

Reply:
Do you mean simplifying it manually or simplifying the generated code ?